File: Java/238.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] arr1 = new int[n];
        int[] arr2 = new int[n];
        int[] lst = new int[n];
        arr1[0] = 1;
        arr2[n-1] = 1;
        for(int i=1;i<n;i++){
            arr1[i] = arr1[i-1] * nums[i-1];
        }
        for(int i=n-2;i>-1;i--){
            arr2[i] = arr2[i+1] * nums[i+1];
        }
        for(int i=0;i<n;i++){
            lst[i] = arr1[i] * arr2[i];
        }
        return lst;
    }
}

File: Python/238.py
class Solution(object):
    def productExceptSelf(self, nums):
        lst1 = [1]
        lst2 = [1]
        lst = []
        n = len(nums)
        for i in range(1,n):
            lst1.append(lst1[-1]*nums[i-1])
        for i in range(n-2,-1,-1):
            lst2.insert(0,lst2[0]*nums[i+1])
        for i in range(n):
            lst.append(lst1[i]*lst2[i])
        return lst